pallets/hsm/src/lib.rs

```diff
@@ -979,12 +996,12 @@ where
 	/// 6. Updating the amount of Hollar received in this block
 	///
 	/// Returns the final Hollar and collateral amounts traded.
-	#[require_transactional]
```

Comment:
in this case, can we have some undesired state-related side effects because we removed require_transactional?

Reply:
reworked. it is better now